app/domain/authentication/authn_oidc/pkce_support_feature/data_objects/authenticator.rb

```diff
@@ -18,6 +18,26 @@ class Authenticator
             :response_type
           )
 
+          # This constant defines a list of annotations that can be included
+          # on a webservice as custom configuration.
+          #
+          # The 'token/ttl' annotation value must be a properly formatted
+          # ISO 8601 Duration. For example, the following annotation included on
+          # an AuthnOIDC V2 webservice will configure the authenticator instance
+          # to distribute tokens that are valid for 30 minutes:
+          #   token/ttl: 'PT30M'
+          OPTIONAL_ANNOTATIONS = %w[token/ttl]
+
+          # These constants are used by the Authenticator class to enforce
+          # restrictions on the custom token TTL, regardless of source.
+          #
+          # Currently, the TokenFactory class enforces default (same as below)
+          # and maximum (5 hours) TTL values. With these constants defined here,
+          # different authn flavors can be subject to unique restrictions.
+          TOKEN_TTL_MIN = 5.minutes
```

Comment:
Instead of forcing the Annotation class to have an awareness of a Conjur token (and that token's TTL), let's push all this into the `TokenFactory` class. Something like this should allow us us to allow us to refactor the `TokenFactory` class to something like:

```ruby
class TokenFactory < Dry::Struct

  # TODO: this error should be moved to the main `errors.rb` file
  NoSigningKey = ::Util::ErrorClass.new(
    "Signing key not found for account '{0}'")

  attribute :slosilo, ::Types::Any.default{ Slosilo }

  MAXIMUM_AUTHENTICATION_TOKEN_EXPIRATION = 5.hours
  MINIMUM_AUTHENTICATION_TOKEN_EXPIRATION = 2.minutes

  def signing_key(account)
    slosilo["authn:#{account}".to_sym] || raise(NoSigningKey, account)
  end

  def signed_token(
    account:,
    username:,
    host_ttl: Rails.application.config.conjur_config.host_authorization_token_ttl,
    user_ttl: Rails.application.config.conjur_config.user_authorization_token_ttl
  )
    signing_key(account).issue_jwt(
      sub: username,
      exp: Time.now + offset(
        ttl: username.starts_with?('host/') ? host_ttl : user_ttl
      )
    )
  end

  def offset(ttl:)
    offset = parse_ttl(ttl: ttl)
    return MAXIMUM_AUTHENTICATION_TOKEN_EXPIRATION if offset > MAXIMUM_AUTHENTICATION_TOKEN_EXPIRATION
    return MINIMUM_AUTHENTICATION_TOKEN_EXPIRATION if offset < MINIMUM_AUTHENTICATION_TOKEN_EXPIRATION

    offset
  end

  def parse_ttl(ttl:)
    # If TTL is an integer
    return ttl.to_i if ttl.to_i.to_s == ttl.to_s

    # If TTL is ISO 8601 Duration
    begin
      ActiveSupport::Duration.parse(ttl).to_i
    rescue ActiveSupport::Duration::ISO8601Parser::ParsingError
      # Attempt to coerce string into integer
      ttl.to_s.to_i
    end
  end
end
```